### PR TITLE
usb: device_next: Fix string descriptors response

### LINKED ISSUE #73073: samples: subsys: usb: hid-keyboard: Assertion failure on USB cable attach when CONFIG_ASSERT=y
**Describe the bug**
An assertion failure is always observed when USB cable is attached. Seems like bug in USB next stack.

**To Reproduce**
I replicated the problem on `nrf52840dk/nrf52840` (built sample with `CONFIG_ASSERT=y`). I think issue also affects other boards.

1. `west build -b nrf52840dk/nrf52840 -p -- -DCONFIG_ASSERT=y`
2. `west flash --erase`
3. Attach USB cable
4. See assertion failure

**Expected behavior**
Assertion failure should not happen.

**Impact**
The assertion also replicates for our other applications. This makes the applications unable to use USB next stack.

**Logs and console output**
```
*** Booting Zephyr OS build v3.6.0-4317-g8716b6a9004a ***
[00:00:00.462,249] <inf> main: HID keyboard sample is initialized
ASSERTION FAIL [ascii7_str[i] > 0x1F && ascii7_str[i] < 0x7F] @ WEST_TOPDIR/zephyr/subsys/usb/device_next/usbd_ch9.c:592
        Only printable ascii-7 characters are allowed in USB string descriptors
[00:00:00.763,977] <err> os: r0/a1:  0x00000004  r1/a2:  0x00000250  r2/a3:  0x00000002
[00:00:00.764,007] <err> os: r3/a4:  0x00000004 r12/ip:  0x00000000 r14/lr:  0x0000418d
[00:00:00.764,007] <err> os:  xpsr:  0x01000000
[00:00:00.764,038] <err> os: Faulting instruction address (r15/pc): 0x0000e6a6
[00:00:00.764,068] <err> os: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
[00:00:00.764,099] <err> os: Current thread: 0x20000aa0 (unknown)
[00:00:01.808,013] <err> os: Halting system
```

**Environment (please complete the following information):**
 - OS: Ubuntu Linux
 - Toolchain: Zephyr SDK 0.16.5
 - Zephyr main branch (`ec0e2d6308680203bb58275062b79105016ea1b1`)

### PR DESCRIPTION
**Pull request by tmon-nordic**
Commit 2f31ee63b5c5 ("usb: device_next: convert ASCII7 strings to UTF16LE on the fly") made string descriptors respond with twice as much of the actual string data.

Fix the issue by taking into account that USB string descriptor length is already multiplied by two. Additionally, make it possible to return odd number of bytes if host requested so.

Fixes: #73073

### CODE REVIEW

**Review by MarekPieta — APPROVED** [review]
The assertion failures seem fixed. Thanks!

**Review by jfischer-no — DISMISSED** [review]

**Review by jfischer-no — APPROVED** [review]

**Comment by tmon-nordic** [comment]
> Hi @jfischer-no, Can we merge it and close my issue?

It is up to someone with merge rights, not up to @jfischer-no

**Review by carlescufi — APPROVED** [review]

**Pull request merged**
